Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -133,13 +133,17 @@ public struct PackageBuilder {
     /// The path of the package.
     private let packagePath: AbsolutePath
 
+    /// The filesystem package builder will run on.
+    private let fileSystem: FileSystem
+
     /// Create a builder for the given manifest and package `path`.
     ///
     /// - Parameters:
     ///   - path: The root path of the package.
-    public init(manifest: Manifest, path: AbsolutePath) {
+    public init(manifest: Manifest, path: AbsolutePath, fileSystem: FileSystem = localFileSystem) {
         self.manifest = manifest
         self.packagePath = path
+        self.fileSystem = fileSystem
```

Comment:
Eventually I think the PackageBuilder should assume it is given a rerooted one, but that might need to be staged in.

The use case I am working towards is that we can run `PackageBuilder` directly against a git repository that doesn't even exist in the file system.